Update gherkin/reference.md (#420)
cucumber/docs: content/docs/gherkin/reference.md

title: Gherkin Reference
subtitle: "Cucumber syntax: Given, When, Then"
---
Gherkin uses a set of special [keywords](#keywords) to give structure and meaning to
executable specifications. Each keyword is translated to many spoken languages;
in this reference we'll use English.

Most lines in a Gherkin document start with one of the [keywords](#keywords).

> A comment must start on a new line. Comment lines are allowed anywhere in the file. They begin with zero or more spaces,

Reviewer: this sentence contradicts itself.

```
When I do something # a comment
```
As an example is ambiguous by this statement.

If you leave it as mentioned before. Or perhaps amend it to something like `Comments are only permitted at the start of a new line, anywhere in the feature file`

> A comment must start on a new line. Comment lines are allowed anywhere in the file. They begin with zero or more spaces,

Reviewer 2: ```suggestion
Comments are only permitted at the start of a new line, anywhere in the feature file. They begin with zero or more spaces,
```

The keyword `Scenario` is a synonym of the keyword `Example`.

You can have as many steps as you like, but we recommend you keep the number at 3-5 per example.

> Having too many steps within examples will cause them to lose their expressive power as specification and documentation.

Reviewer 2: ```suggestion
Having too many steps in an example, will cause it to lose their expressive power as specification and documentation.
```

compare the *actual* outcome (what the system actually does) to the *expected* outcome
(what the step says the system is supposed to do).

> An outcome *should* be on an **observable** output. That is, something that comes *out* of the system (report, user interface, message), and not a behaviour deeply buried inside the system (like a minor change to a database).

Reviewer 2: ```suggestion
An outcome *should* be on an **observable** output. That is, something that comes *out* of the system (report, user interface, message), and not a behaviour deeply buried inside the system (like a record in a database).
```

While it might be tempting to implement `Then` steps to look in the database - resist that temptation!

> You should only verify an outcome that is observable for the user (or external system), and changes to a database are usually not.

Reviewer 2: Nice catch, thanks!
